### PR TITLE
Doc update for api, jupyter, visualization, and test

### PR DESCRIPTION
**Pull request by jyu00**
### Summary
Doc updates for
- api
- jupyter
- visualization
- test

Also changed `apidoc` to use short description instead of package name to match terra.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-####################################
-Qiskit IBM Q Provider documentation
-####################################
+#########################################
+Qiskit IBM Quantum Provider documentation
```
Comment (luisg5) [inline docs/index.rst]: IBM Quantum Experience Provider?

File: test/ibmq/__init__.py [hunk 2]
```diff
@@ -12,4 +12,4 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Qiskit IBMQ integration tests."""
+"""Tests for IBM Quantum Provider."""
```
Comment (luisg5) [inline test/ibmq/__init__.py]: IBM Quantum Experience Provider?

File: qiskit/providers/ibmq/api/rest/backend.py [hunk 3]
```diff
@@ -35,8 +35,8 @@ def __init__(self, session: RetrySession, backend_name: str) -> None:
         """Backend constructor.
 
         Args:
-            session: session to be used in the adaptor.
-            backend_name: name of the backend.
+            session: Session to be used in the adaptor.
```
[the remark this answers is not in the document]
Comment (jyu00) [reply qiskit/providers/ibmq/api/rest/backend.py]: According to Google, `adaptor` is actually a valid spelling. It's just less common in the US.

File: docs/index.rst (hunk 1 above)
Comment (jyu00) [reply docs/index.rst]: So we decided on just `IBM Quantum Provider` for now, to match `ibmq-provider` (see PR #542 )

File: test/ibmq/__init__.py (hunk 2 above)
Comment (jyu00) [reply test/ibmq/__init__.py]: See above.

**Pull request merged**